AK/Math.h

```diff
@@ -984,11 +984,25 @@ using Hyperbolic::cosh;
 using Hyperbolic::sinh;
 using Hyperbolic::tanh;
 
+// Calculate x^y with fast exponentiation when the power is an integer.
+template<FloatingPoint F, Integral I>
+constexpr F pow_int(F x, I y)
```

Comment:
As this only works for Unsigned y, maybe we could enforce that

Reply:
Yeah, immediately noticed after pushing that that was wrong. It won’t break at the moment since we check against i64 bounds, which are smaller than u64 bounds in the positive.